CMakeLists.txt

```diff
@@ -963,8 +963,15 @@ else()
   set(PC_INC_INSTALL_DIR "${CMAKE_INSTALL_INCLUDEDIR}")
 endif()
 
+if(APPLE)
+  option(WITH_RPATH "Enable RPATH for shared library" OFF)
```

Comment:
Why have an option? Just allow it always.

Reply:
Because it would break installation to system directory. You can't just change `CMAKE_INSTALL_LIBDIR` to alter RPATH, because the default might match the new value as it's not guaranteed to be absolute path.

Reply:
Having an option allows backwards compatibility (if someone depended on this). But I think that default should be ON.

Reply:
The issue with having it ON by default is that there is no way to recover the full RPATH from the information in the cache because the cache isn't built at this stage yet... so the RPATH would be incorrect. With `configure` there is no issue as we don't use cache at all.

It's a lot more common for the `CMAKE_INSTALL_LIBDIR` to contain just `lib` than to contain full path like `/usr/local/lib`. As there is no cache yet, we can't use `CMAKE_INSTALL_FULL_LIBDIR` to determine what the full path would be, as it isn't updated before first `install()` call.

Reply:
In ITK, we will want to use RPATH by default. I assume the problem with @kislinsk's suggestion (remove custom code) is that configure has slightly different behavior?

Reply:
The problem is that cmake doesn't like relative paths, so `./lib` must be specified as `/lib` even though the two refer to two distinct paths.

Reply:
It does seem pretty cludgy, but I guess if anyone knows of a cleaner way to do this, they can open a PR.

Reply:
cmake has a lot more limitations than configure script and I'm pretty sure cmake developers are not going to fix every limitation that is by design. I tried whole night different options and either it failed CI tests, or it didn't work as expected.

Only limitation in configure script and its Makefiles affecting this PR is that `DESTDIR` must end with `/´, otherwise relative paths don't work at all...

Reply:
@bradking Do you have some suggestion for CMake build system?